Makefile

```diff
@@ -1,52 +1,191 @@
+APP_NAME    = $(shell basename -s .git `git config --get remote.origin.url`)
+# Get current commit
+APP_COMMIT  = $(shell git log --pretty=format:'%h' -n 1)
+# Check if we are in a release tag, if yes use it as app version.
+APP_VERSION = $(shell git describe --abbrev=0 --exact-match --tags > /dev/null 2>&1 || echo dev-$(APP_COMMIT))
+BUILD_DATE  = $(shell date -u)
+
+## General Configuration Variables
+# We don't need make's built-in rules.
+MAKEFLAGS += --no-builtin-rules
+# Be pedantic about undefined variables.
+MAKEFLAGS += --warn-undefined-variables
+# Set help as default target
+.DEFAULT_GOAL := help
+
+# ====================================================================================
+# Colors
+
+BLUE         := $(shell printf "\033[34m")
+YELLOW       := $(shell printf "\033[33m")
+RED          := $(shell printf "\033[31m")
+GREEN        := $(shell printf "\033[32m")
+CNone        := $(shell printf "\033[0m")
+
+# ====================================================================================
+# Logger
+
+TIME_LONG	= `date +%Y-%m-%d' '%H:%M:%S`
+TIME_SHORT	= `date +%H:%M:%S`
+TIME		= $(TIME_SHORT)
+
+INFO	= echo ${TIME} ${BLUE}[ .. ]${CNone}
+WARN	= echo ${TIME} ${YELLOW}[WARN]${CNone}
+ERR		= echo ${TIME} ${RED}[FAIL]${CNone}
+OK		= echo ${TIME} ${GREEN}[ OK ]${CNone}
+FAIL	= (echo ${TIME} ${RED}[FAIL]${CNone} && false)
+
+# ====================================================================================
+# Targets
+
+help: ## to get help
+	@echo "Usage:"
+	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) |\
+	awk 'BEGIN {FS = ":.*?## "}; {printf "make \033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-.PHONY: build
-build:
-	mkdir -p dist
-	env GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -o dist/rtcd -ldflags '$(LDFLAGS)' -mod=readonly -trimpath ./cmd/rtcd
+.PHONY: lint
+lint: go-lint go-mod-check docker-lint  ## to lint all
+
+.PHONY: test
+test: go-test  ## to test all
 
 .PHONY: docker-build
-docker-build:
-	@if ! [ -x "$$(command -v docker)" ]; then \
-		echo "Docker is not installed. Please see https://docs.docker.com/get-docker for installation instructions."; \
-		exit 127; \
-	fi; \
+docker-build: go-build ## to build the docker image
+	@$(INFO) Performing Docker build...
+	@docker build -f ${CONFIG_DOCKERFILE} . \
+	-t ${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Performing Docker build ${APP_NAME}:${APP_VERSION}
+
+.PHONY: docker-push
+docker-push: ## to push the docker image
+	@$(INFO) Pushing to registry...
+	@docker tag ${APP_NAME}:${APP_VERSION} $(CONFIG_IMAGE_REGISTRY)/${CONFIG_DOCKER_REPO}/${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@docker push $(CONFIG_IMAGE_REGISTRY)/${CONFIG_DOCKER_REPO}/${APP_NAME}:${APP_VERSION} || ${FAIL}
+	@$(OK) Pushing to registry $(CONFIG_IMAGE_REGISTRY)/${CONFIG_DOCKER_REPO}/${APP_NAME}:${APP_VERSION}
+
+.PHONY: docker-sbom
+docker-sbom: ## to print a sbom report
+	@$(INFO) Performing Docker sbom report...
+	@docker sbom ${APP_NAME}:${APP_VERSION} || ${FAIL}
```

Comment:
is this `SPDX`? btw I think there is a flag `--format spdx-json` as JSON will be useful  and upload in artifacts of Github actions for example?

Reply:
Good point! Noting this down and will add it in the next iteration !